src/coreclr/tools/Common/Compiler/Logger.cs

```diff
@@ -27,46 +29,56 @@ public class Logger
         private readonly HashSet<string> _trimWarnedAssemblies = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
         private readonly HashSet<string> _aotWarnedAssemblies = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
-        public static Logger Null = new Logger(TextWriter.Null, false);
+        public static Logger Null = new Logger(new TextLogWriter(TextWriter.Null), false);
 
-        public TextWriter Writer { get; }
+        public TextWriter Writer => _logWriter.Writer;
```

Comment:
On my phone so just asking here - would it be possible to migrate uses of this to LogMessage. Then maybe ILogWriter doesn't need the Writer property?